internal/core/src/common/VectorArray.h

```diff
@@ -0,0 +1,371 @@
+        : data_(data),
+          dim_(dim),
+          length_(len),
+          size_(size),
+          element_type_(element_type) {
+    }
+
+    template <typename VectorElement>
+    VectorElement*
+    get_data(const int index) const {
+        AssertInfo(index >= 0 && index < length_,
+                   "index out of range, index={}, length={}",
+                   index,
+                   length_);
+        switch (element_type_) {
+            case DataType::VECTOR_FLOAT: {
+                return static_cast<VectorElement*>(
+                    reinterpret_cast<float*>(data_) + index * dim_);
+            }
+            default: {
+                // TODO(SpadeA): add other vector types
+                PanicInfo(NotImplemented,
+                          "Not implemented vector type: {}",
+                          element_type_);
+            }
+        }
+    }
+
+    VectorFieldProto
+    output_data() const {
+        VectorFieldProto vector_array;
+        vector_array.set_dim(dim_);
+        switch (element_type_) {
+            case DataType::VECTOR_FLOAT: {
+                auto data = reinterpret_cast<const float*>(data_);
+                vector_array.mutable_float_vector()->mutable_data()->Add(
+                    data, data + length_ * dim_);
+                break;
+            }
+            default: {
+                // TODO(SpadeA): add other vector types
+                PanicInfo(NotImplemented,
+                          "Not implemented vector type: {}",
+                          element_type_);
+            }
+        }
+        return vector_array;
+    }
+
+    bool
+    is_same_array(const VectorFieldProto& vector_field) {
+        switch (element_type_) {
+            case DataType::VECTOR_FLOAT: {
+                if (vector_field.data_case() != VectorFieldProto::kFloatVector) {
+                    return false;
+                }
+                
+                if (length_ !=
+                    vector_field.float_vector().data().size() / dim_) {
+                    return false;
+                }
+
+                if (length_ == 0) {
+                    return true;
+                }
+
+                for (int i = 0; i < length_; ++i) {
+                    auto a = get_data<float>(i);
+                    auto b =
+                        vector_field.float_vector().data().begin() + i * dim_;
+                    for (int j = 0; j < dim_; ++j) {
+                        if (a[j] != b[j]) {
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+            default: {
+                // TODO(SpadeA): add other vector types
+                PanicInfo(NotImplemented,
+                          "Not implemented vector type: {}",
+                          element_type_);
+            }
+        }
+    }
+
+ private:
+    char* data_{nullptr};
+    int64_t dim_ = 0;
+    int length_ = 0;
```

Comment:
meaning of member length_ and size_ are ambiguous, add comments.

Reply:
ok